MTPot.py

```diff
@@ -2,29 +2,98 @@
+import socket
 
+default_timeout = 60 #Use to timeout the connection
 COMMANDS = {}
 COMMANDS_EXECUTED = {}
+OVERWRITE_COMMANDS = {} #Use to overwrite default telnet command behavior crahsing the handler (e.g. 'help')
+OVERWRITE_COMMANDS_LIST = ["help"] #Don't forget to update the list when adding new commands
 BUSY_BOX = "/bin/busybox"
 MIRAI_SCANNER_COMMANDS = ["shell", "sh", "enable"]
 FINGERPRINTED_IPS = []
 
 honey_logger = logging.getLogger("HoneyTelnet")
 syslogger = None
 config = None
+custom_pool = None
+
+
+'''
+    An extension of the gevent pool.
+    If this pool becomes full, it drops oldest connections instead of waiting their end.
+'''
+class CustomPool(gevent.pool.Pool):
+
+    def __init__(self, size=None, greenlet_class=None):
+        self.open_connection = []   #FIFO for connection
+        self.open_connection_dico_ip = {} #2-way dico
+        self.open_connection_dico_green = {} #2-way dico
+        gevent.pool.Pool.__init__(self, size+1, greenlet_class) #+1 to avoid the semaphore
+
+    # Add the greenlet to the pool
+    def add(self, greenlet):
+        source = greenlet.args[2][1][0] + ':' + str(greenlet.args[2][1][1])
+        socket = greenlet.args[2][0]
+
+        # With 1, we avoid the wait caused by the semaphore
+        if self.free_count() < 2:
+            # /!\ pool full, untracking oldest greenlet /!\
+            oldest_source = self.open_connection[0]
+            oldest_greenlet = self.open_connection_dico_ip[oldest_source]
+
+            #kill the greenlet, this also close its associated socket
+            self.killone(oldest_greenlet, block=False)
+
+        #Add the connection to the dicos
+        self.open_connection.append(source)
+        self.open_connection_dico_ip[source] = greenlet
+        self.open_connection_dico_green[str(greenlet)] = source
+        gevent.pool.Pool.add(self, greenlet)
+
+    # discard the greenlet, free one slot of the pool
+    def _discard(self, greenlet):
+        to_del_greenlet = str(greenlet)
+        to_del_source = self.open_connection_dico_green[to_del_greenlet]
+        gevent.pool.Pool._discard(self, greenlet)
+
+        #cleaning dicos
+        del self.open_connection_dico_ip[to_del_source]
+        del self.open_connection_dico_green[to_del_greenlet]
+        self.open_connection.remove(to_del_source)
+
+    def print_pool_info(self):
+        print 'pool size', self.free_count()-1
+        #print 'open connection', self.open_connection
+        #print 'dico ip', self.open_connection_dico_ip
+        #print 'dico green', self.open_connection_dico_green
+
+    def remove_connection(self, to_del_source):
+        to_del_greenlet = self.open_connection_dico_ip[to_del_source]
+        self._discard(to_del_greenlet)
+
 
 class MyTelnetHandler(TelnetHandler):
     WELCOME = 'welcome'
     PROMPT = ">"
     authNeedUser = True
     authNeedPass = True
 
+    @command(OVERWRITE_COMMANDS_LIST)
+    def shell_respond2(self, params):
```

Comment:
I'd change the name of the function to something like - telnet_commands_respond (this function doesn't handle shell commands)